app/bundles/DynamicContentBundle/Entity/DynamicContentLeadData.php

```diff
@@ -66,7 +66,7 @@ public static function loadMetadata(ORM\ClassMetadata $metadata)
 
         $builder->createManyToOne('dynamicContent', 'DynamicContent')
             ->inversedBy('id')
-            ->addJoinColumn('dynamic_content_id', 'id')
+            ->addJoinColumn('dynamic_content_id', 'id', true, false, 'CASCADE')
```

Comment:
According to [this migration](https://github.com/mautic/mautic/blob/dc2ba012c34921facf6d8f12d334e1f15db4c622/app/migrations/Version20160624032452.php#L57), the last parameter should be `'SET NULL'` instead of `'CASCADE'`. In my understanding, CASCADE will remove the joined records. SET NULL will set the dynamic_content_id in the dynamic_content_lead_data to NULL. This way you won't lose the records which can be still used for global statistics or user actions.